infrastructure/src/main/java/org/corfudb/infrastructure/log/StreamLogFiles.java

```diff
@@ -84,14 +86,46 @@
     private Map<String, SegmentHandle> writeChannels;
     private Set<FileChannel> channelsToSync;
     private MultiReadWriteLock segmentLocks = new MultiReadWriteLock();
+    final private ServerContext serverContext;
+    final private AtomicLong globalTail = new AtomicLong(0L);
+    final private AtomicLong tempGlobalTail = new AtomicLong(0L);
```

Comment:
this name is not very intuitive. please change it to something better. Also please add comments for these fields.